python/hail/type.py

```diff
@@ -1,12 +1,582 @@
+from hail.java import scala_object, scala_package_object, Env
+from hail.representation import Variant, AltAllele, Genotype, Locus, Interval, Struct
+
+
+class TypeCheckError(Exception):
+    """
+    Error thrown at mismatch between expected and supplied python types.
+
+    :param str message: Error message
+    """
+
+    def __init__(self, message):
+        self.msg = message
+        super(TypeCheckError).__init__(TypeCheckError)
+
+    def __str__(self):
+        return self.msg
+
 
 class Type(object):
-    """Type of values."""
-    
+    """
+    Hail type class used for annotations and expression language.
+
+    :param jtype: equivalent java type
```

Comment:
Again, I suggest we don't document constructors with java objects the user can't/shouldn't call.